sdk/python/kfp/components/_naming.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = [
+    '_normalize_identifier_name',
+    '_sanitize_kubernetes_resource_name',
+    '_sanitize_python_function_name',
+    '_sanitize_file_name',
+    '_generate_unique_suffix',
+    '_make_name_unique_by_adding_index',
+]
+
+
+import re
+import sys
+
+
+def _normalize_identifier_name(name):
```

Comment:
Unit test for those utility methods? I know those methods are not new ones. Do we have existing tests covering them?

Reply:
I have a plan to add the tests in a separate PR.